scripts/builtin/imputeByKNN.dml

```diff
@@ -0,0 +1,208 @@
+#-------------------------------------------------------------
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+#-------------------------------------------------------------
+
+# Impute the data by using KNN-algorithm and finding the nearest neighbor using euclidean distance
+# Currently only for single column with multiple missing values and top 1 neighbor
+#
+# INPUT:
+# -------------------------------------------------------------------------------------
+# data    Data Matrix (numerical)
+# method  methods of calculating the KNN-algorith depending on the size of data and missing values
```

Comment:
Include the valid method names in the documentation, where you pick meaningful names other than method2 and method3

Reply:
Is it fine to use "default", "small", and "large" or do you have other suggestions?

Reply:
no, give meaningful names: such as dist, dist_missing, dist_sample - and they need to be explained in the header documentation.